lib/AtomicCache.php

```diff
@@ -143,31 +286,26 @@ public function get(string $key): Promise
     }
 
     /**
-     * The lock is obtained for the key before setting the value.
+     * Returns the cached value for the key or the given default value if the key does not exist.
      *
-     * @param $key   string Cache key.
-     * @param $value mixed Value to cache.
-     * @param $ttl   int Timeout in seconds. The default `null` $ttl value indicates no timeout. Values less than 0 MUST
-     *               throw an \Error.
+     * @param string $key Cache key.
+     * @param mixed  $default Default value returned if the key does not exist.
      *
-     * @return Promise<void> Resolves either successfully or fails with a CacheException on failure.
+     * @return Promise<mixed|null>
      *
      * @throws CacheException
      * @throws SerializationException
-     *
-     * @see SerializedCache::set()
      */
-    public function set(string $key, $value, ?int $ttl = null): Promise
+    public function getOrDefault(string $key, $default): Promise
```

Comment:
I think we can just merge this into `get`, so make `get(string $key, $default = null)`.